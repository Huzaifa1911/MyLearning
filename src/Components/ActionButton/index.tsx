import {StyleSheet, TouchableOpacity} from 'react-native';
import React from 'react';

import AppText from '../Common/AppText';
import {AppTheme} from '../../Assets/Theme';

interface IActionButtonProps {
  title: string;
  onPress: () => void;
}

const ActionButton = ({onPress, title}: IActionButtonProps) => {
  return (
    <TouchableOpacity
      style={styles.container}
      activeOpacity={0.6}
      onPress={onPress}>
      <AppText size={14} fontWeight="600">
        {title}
      </AppText>
    </TouchableOpacity>
  );
};

export default ActionButton;

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 120,
    width: 140,
    borderRadius: 8,
    backgroundColor: AppTheme.colors.white,
    borderWidth: 0.3,
    marginTop: 10,
  },
});
