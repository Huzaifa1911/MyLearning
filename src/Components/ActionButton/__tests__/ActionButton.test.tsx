/* eslint-disable react/react-in-jsx-scope */
import {fireEvent, render} from '@testing-library/react-native';

import ActionButton from '..';

describe('Action Button', () => {
  it('Renders Correctly', () => {
    const tree = render(<ActionButton title="Title" onPress={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  // Checking onPress
  it('Calls onPress', async () => {
    // Mocking onPress, so that we will able to test.
    const onPress = jest.fn();
    const {getByTestId} = render(
      <ActionButton title="ActionButton" onPress={onPress} />,
    );

    const button = getByTestId('ActionButton');
    fireEvent.press(button);
    expect(onPress).toHaveBeenCalled();
  });
});
