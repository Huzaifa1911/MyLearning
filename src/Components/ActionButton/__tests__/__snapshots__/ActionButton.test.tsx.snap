// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Button Renders Correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "borderWidth": 0.3,
      "height": 120,
      "justifyContent": "center",
      "marginTop": 10,
      "opacity": 1,
      "width": 140,
    }
  }
  testID="ActionButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flex": undefined,
          "marginBottom": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": 5,
          "marginTop": undefined,
          "marginVertical": undefined,
        }
      }
    />
    <Text
      style={
        {
          "color": "#18234C",
          "fontSize": 14,
          "fontWeight": "600",
          "textAlign": "left",
          "textDecorationLine": "none",
          "textTransform": "none",
        }
      }
      testID="title-Title"
    >
      Title
    </Text>
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flex": undefined,
          "marginBottom": undefined,
          "marginHorizontal": undefined,
          "marginLeft": 5,
          "marginRight": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
        }
      }
    />
  </View>
</View>
`;
